lib/validate_url.rb

```diff
@@ -19,8 +21,10 @@ def validate_each(record, attribute, value)
         schemes = [*options.fetch(:schemes)].map(&:to_s)
         begin
           uri = URI.parse(value)
-          unless uri && uri.host && schemes.include?(uri.scheme) && (!options.fetch(:no_local) || uri.host.include?('.'))
-            record.errors.add(attribute, :url, filtered_options(value))
+          validate_suffix = !options.fetch(:public_suffix) || (uri && uri.host && PublicSuffix.valid?(uri.host, :default_rule => nil))
+          validate_no_local = !options.fetch(:no_local) || uri.host.include?('.')
```

Comment:
This breaks urls which don't have a host e.g. `www.example.com` because `host` returns `nil`

Reply:
Oh my bad, will PR a hotfix.